src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -425,13 +419,19 @@ private bool IsUnfixed(int methodTypeParameterIndex)
             return (object)_fixedResults[methodTypeParameterIndex] == null;
         }
 
-        private bool IsUnfixedTypeParameter(TypeSymbol type)
+        private bool IsUnfixedTypeParameter(TypeSymbolWithAnnotations source, TypeSymbolWithAnnotations target)
```

Comment:
As we discussed, let's return this method to original signature.

Reply:
Yes, this means we need to check `{Exact,LowerBound,UpperBound}NullableInference` first, before  `{Exact,LowerBound,UpperBound}TypeParameterInference` in each case.